src/python_testing/TC_IDM_4_2.py

```diff
@@ -0,0 +1,589 @@
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import copy
+import logging
+import random
+import time
+
+import chip.clusters as Clusters
+from chip.ChipDeviceCtrl import ChipDeviceController
+from chip.clusters.Attribute import AttributePath, TypedAttributePath
+from chip.exceptions import ChipStackError
+from chip.interaction_model import Status
+from matter_testing_support import MatterBaseTest, async_test_body, default_matter_test_main
+from mobly import asserts
+
+
+class TC_IDM_4_2(MatterBaseTest):
+
+    async def write_acl(self, acl):
+        result = await self.default_controller.WriteAttribute(self.dut_node_id, [(0, Clusters.AccessControl.Attributes.Acl(acl))])
+        asserts.assert_equal(result[0].Status, Status.Success, "ACL write failed")
+
+    async def get_descriptor_server_list(self, ctrl):
+        return await self.read_single_attribute_check_success(
+            endpoint=0,
+            dev_ctrl=ctrl,
+            cluster=Clusters.Descriptor,
+            attribute=Clusters.Descriptor.Attributes.ServerList
+        )
+
+    async def get_idle_mode_duration(self, ctrl):
+        return await self.read_single_attribute_check_success(
+            endpoint=0,
+            dev_ctrl=ctrl,
+            cluster=Clusters.IcdManagement,
+            attribute=Clusters.IcdManagement.Attributes.IdleModeDuration
+        )
+
+    def verify_attribute_data(self, sub, cluster, attribute, ep=0):
+        sub_attrs = sub
+        if isinstance(sub, Clusters.Attribute.SubscriptionTransaction):
+            sub_attrs = sub.GetAttributes()
+
+        asserts.assert_true(ep in sub_attrs, "Must have read endpoint %s data" % ep)
+        asserts.assert_true(cluster in sub_attrs[0], "Must have read %s cluster data" % cluster.__name__)
+        asserts.assert_true(attribute in sub_attrs[0][cluster],
+                            "Must have read back attribute %s" % attribute.__name__)
+
+    def get_attribute_from_sub_dict(self, sub, cluster, attribute, ep=0):
+        return sub[ep][cluster][attribute]
+
+    def get_typed_attribute_path(self, attribute, ep=0):
+        return TypedAttributePath(
+            Path=AttributePath(
+                EndpointId=ep,
+                Attribute=attribute
+            )
+        )
+
+    async def get_dut_acl(self):
+        sub = await self.default_controller.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=[(0, Clusters.AccessControl.Attributes.Acl)],
+            keepSubscriptions=False,
+            fabricFiltered=True
+        )
+
+        acl_list = self.get_attribute_from_sub_dict(
+            sub=sub,
+            cluster=Clusters.AccessControl,
+            attribute=Clusters.AccessControl.Attributes.Acl
+        )
+
+        return acl_list
+
+    def is_uint32(self, var):
```

Comment:
nit: we do not know if this is uint32, but rather if this is a valid uint32 value. Basically we have the value `123` here being both `is_uint32` and `is_uint16`.

How about naming these `is_valid_uint.._value` ?

These methods do not use `self` so there is not need for them to be instance methods. Could do `@staticmethod`, however maybe top level stand alone functions would be better.

Reply:
Updated.